pkg/util/managedcluster/managedcluster.go

```diff
@@ -42,3 +46,47 @@ func ClientsetFromConfig(cs *api.OpenShiftManagedCluster) (*kubernetes.Clientset
 
 	return kubernetes.NewForConfig(restconfig)
 }
+
+// getKubeconfigFromV1Config takes a v1 config and returns a kubeconfig
+func getKubeconfigFromV1Config(kc *v1.Config) (clientcmd.ClientConfig, error) {
+	var c kapi.Config
+	err := latest.Scheme.Convert(kc, &c, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeconfig := clientcmd.NewDefaultClientConfig(c, &clientcmd.ConfigOverrides{})
+
+	return kubeconfig, nil
+}
+
+// ClientSetFromV1Config takes a v1 config and returns a Clientset
+func ClientSetFromV1Config(ctx context.Context, config *v1.Config) (*kubernetes.Clientset, error) {
+	kubeconfig, err := getKubeconfigFromV1Config(config)
+	if err != nil {
+		return nil, err
+	}
+
+	restconfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	t, err := rest.TransportFor(restconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for the healthz to be 200 status
+	err = wait.ForHTTPStatusOk(ctx, t, restconfig.Host+"/healthz")
```

Comment:
The healthz check is an additional step that should only be required when the cluster is coming up, therefore it shouldn't really be moved into a general package.  Maybe a separate exported function here which returns the restconfig to the caller.  The caller can get the restconfig, poll for /healthz, then get the ClientSet.